dist/validate-release.sh

```
@@ -21,38 +21,44 @@
 # 3. Compile the source package & run server & toolchain
 # 4. Run server & toolchain in binary package
 
-URL_PREFIX="https://dist.apache.org/repos/dist/dev/incubator/hugegraph/"
+set -e
+
 # release version (input by committer)
-RELEASE_VERSION=$1
-JAVA_VERSION=$2
+RELEASE_VERSION=$1 # like 1.2.0
+JAVA_VERSION=$2 # like 8
 USER=$3
+
+# this path is just valid in release-ing progress
+SVN_URL_PREFIX="https://dist.apache.org/repos/dist/dev/incubator/hugegraph"
+
 # git release branch (check it carefully)
 #GIT_BRANCH="release-${RELEASE_VERSION}"
 
-RELEASE_VERSION=${RELEASE_VERSION:?"Please input the release version behind script"}
+RELEASE_VERSION=${RELEASE_VERSION:?"Please input the release version, like 1.2.0"}
+USER=${USER-"imbajin"}
 WORK_DIR=$(
-  cd "$(dirname "$0")" || exit
+  cd "$(dirname "$0")"
   pwd
 )
 
-cd "${WORK_DIR}" || exit
+cd "${WORK_DIR}"
 echo "Current work dir: $(pwd)"
 
 ################################
 # Step 1: Download SVN Sources #
 ################################
-rm -rf "$WORK_DIR"/dist/"$RELEASE_VERSION"
-svn co ${URL_PREFIX}/"$RELEASE_VERSION" "$WORK_DIR"/dist/"$RELEASE_VERSION"
+rm -rf "${WORK_DIR}/dist/${RELEASE_VERSION}"
+mkdir -p "${WORK_DIR}/dist/${RELEASE_VERSION}"
+cd "${WORK_DIR}/dist/${RELEASE_VERSION}"
+svn co "${SVN_URL_PREFIX}/${RELEASE_VERSION}" .
 
 ##################################################
 # Step 2: Check Environment & Import Public Keys #
 ##################################################
-cd "$WORK_DIR"/dist/"$RELEASE_VERSION" || exit
+shasum --version 1>/dev/null
+gpg --version 1>/dev/null
```

another usage for `exit`, when the step failed, the CI should failed in time (rather than ignore it)

---

ditto